core/identity-hub-did/src/main/java/org/eclipse/edc/identityhub/did/DidDocumentServiceImpl.java

```diff
@@ -181,9 +194,77 @@ public ServiceResult<Void> removeService(String did, String serviceId) {
     }
+
+    @Override
+    public <E extends Event> void on(EventEnvelope<E> eventEnvelope) {
+        var payload = eventEnvelope.getPayload();
+        if (payload instanceof ParticipantContextCreated event) {
+            created(event);
+        } else if (payload instanceof ParticipantContextUpdated event) {
+            updated(event);
+        } else if (payload instanceof ParticipantContextDeleted event) {
+            deleted(event);
+        } else {
+            monitor.warning("KeyPairServiceImpl Received event with unexpected payload type: %s".formatted(payload.getClass()));
+        }
+    }
+
+    private void updated(ParticipantContextUpdated event) {
+        var newState = event.getNewState();
+        var forParticipant = findByParticipantId(event.getParticipantId());
+        var errors = forParticipant
+                .stream()
+                .map(resource -> switch (newState) {
+                    case ACTIVATED -> publish(resource.getDid());
+                    case DEACTIVATED -> unpublish(resource.getDid());
+                    default -> ServiceResult.success();
+                })
+                .filter(AbstractResult::failed)
+                .map(AbstractResult::getFailureDetail)
+                .collect(Collectors.joining(", "));
+
+        if (!errors.isEmpty()) {
+            monitor.warning("Updating DID documents subsequent to updating a ParticipantContext failed: %s".formatted(errors));
+        }
+    }
+
+    private void deleted(ParticipantContextDeleted event) {
+        var participantId = event.getParticipantId();
+        //unpublish and delete all DIDs associated with that participant
+        var errors = findByParticipantId(participantId)
+                .stream()
+                .map(didResource -> unpublish(didResource.getDid())
+                        .compose(u -> deleteById(didResource.getDid())))
+                .map(AbstractResult::getFailureDetail)
+                .collect(Collectors.joining(", "));
+
+        if (!errors.isEmpty()) {
+            monitor.warning("Unpublishing/deleting DID documents subsequent to deleting a ParticipantContext failed: %s".formatted(errors));
+        }
+    }
+
+    private Collection<DidResource> findByParticipantId(String participantId) {
+        return didResourceStore.query(QuerySpec.Builder.newInstance().filter(new Criterion("participantId", "=", participantId)).build());
+    }
+
+
+    private void created(ParticipantContextCreated event) {
+        var manifest = event.getManifest();
+        var doc = DidDocument.Builder.newInstance()
+                .id(manifest.getDid())
+                .service(manifest.getServiceEndpoints().stream().toList())
+                // updating and adding a verification method happens as a result of the KeyPairAddedEvent
+                // .verificationMethod(List.of(VerificationMethod.Builder.newInstance()
+                // .publicKeyJwk(publicKey.toJSONObject())
+                // .build()))
```

Comment:
to remove ?

Reply:
i noticed, that the code add verification methods was actually missing 🤦. Will add that and then remove the commented code here.